src/fsharp/FSharp.Core/prim-types.fs

```diff
@@ -1284,8 +1290,18 @@ namespace Microsoft.FSharp.Core
                  when 'T : uint32 = if (# "clt.un" x y : bool #) then (-1) else (# "cgt.un" x y : int #)
                  when 'T : uint64 = if (# "clt.un" x y : bool #) then (-1) else (# "cgt.un" x y : int #)
                  when 'T : unativeint = if (# "clt.un" x y : bool #) then (-1) else (# "cgt.un" x y : int #)
-                 when 'T : float  = if (# "clt" x y : bool #) then (-1) else (# "cgt" x y : int #)
-                 when 'T : float32 = if (# "clt" x y : bool #) then (-1) else (# "cgt" x y : int #)
+                 when 'T : float  = if   (# "clt" x y : bool #) then (-1)
+                                    elif (# "cgt" x y : bool #) then (1)
+                                    elif (# "ceq" x y : bool #) then (0)
+                                    elif (# "ceq" x x : bool #) then (1)
+                                    elif (# "ceq" y y : bool #) then (-1)
+                                    else (0)
```

Comment:
Reordering the nan comparisons we can eliminate a branch

    when 'T : float  = if   (# "clt" x y : bool #) then (-1)
                       elif (# "cgt" x y : bool #) then (1)
                       elif (# "ceq" x y : bool #) then (0)
                       elif (# "ceq" y y : bool #) then (-1)
                       else (# "ceq" x x : bool #)

It's similar to what was done here previously, but doesn't avoid any extra IL calls and is arguably less clear